From 46881dec5205f39f846289e914eddb8ccdf1171f Mon Sep 17 00:00:00 2001
From: Tanya Finkel <tfinkel@codeaurora.org>
Date: Mon, 13 Oct 2014 15:11:26 +0300
Subject: [PATCH] init.qcom.usb: Set USB composition for DB8094 peripheral
 mode.

For DB8094 the usb peripheral connected to port 1 CI
that support only USB2.

Change-Id: Ia6becd299de02737be2dac21af85fbbe92240213
---
 rootdir/etc/init.qcom.usb.sh | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/rootdir/etc/init.qcom.usb.sh b/rootdir/etc/init.qcom.usb.sh
index 2fd4914..5e05e46 100644
--- a/rootdir/etc/init.qcom.usb.sh
+++ b/rootdir/etc/init.qcom.usb.sh
@@ -30,6 +30,13 @@
 chown -h root.system /sys/devices/platform/msm_hsusb/gadget/wakeup
 chmod -h 220 /sys/devices/platform/msm_hsusb/gadget/wakeup
 
+# Set platform variables
+if [ -f /sys/devices/soc0/hw_platform ]; then
+    soc_hwplatform=`cat /sys/devices/soc0/hw_platform` 2> /dev/null
+else
+    soc_hwplatform=`cat /sys/devices/system/soc/soc0/hw_platform` 2> /dev/null
+fi
+
 #
 # Allow persistent usb charging disabling
 # User needs to set usb charging disabled in persist.usb.chgdisabled
@@ -123,7 +130,11 @@ case "$usb_config" in
 				setprop persist.sys.usb.config diag,serial_smd,rmnet_bam,adb
 			;;
 			"msm8994")
-				setprop persist.sys.usb.config diag,serial_smd,serial_tty,rmnet_ipa,mass_storage,adb
+				if ["$soc_hwplatform" == "Dragon" ]; then
+					setprop persist.sys.usb.config diag,adb
+				else
+					setprop persist.sys.usb.config diag,serial_smd,serial_tty,rmnet_ipa,mass_storage,adb
+				fi
 			;;
 			"ferrum")
 				setprop persist.sys.usb.config diag,adb
-- 
1.8.2.1

